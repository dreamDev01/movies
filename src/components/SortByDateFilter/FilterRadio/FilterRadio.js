import "./FilterRadio.scss";

/*-------FilterRadio----------*/
/*props :          */
/*  - value        */
/*  - label        */
/*  - handleChange */
/*  - checked      */
const FilterRadio = (props) => {
  const { value, label, handleChange, checked } = props;

  return (
    <div className="radio">
      <input
        data-testid={value}
        checked={checked}
        type="radio"
        value={value}
        onChange={handleChange}
      ></input>
      {label}
    </div>
  );
};

export default FilterRadio;
