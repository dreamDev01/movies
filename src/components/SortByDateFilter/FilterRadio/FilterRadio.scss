.radio {
  padding-left: 10px;
  margin: 10px 0 10px 0;
  height: 30px;

  input {
    cursor: pointer;
    margin-top: 5px;
    margin-right: 10px;
  }
}
