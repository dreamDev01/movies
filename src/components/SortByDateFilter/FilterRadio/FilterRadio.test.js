import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import FilterRadio from "./FilterRadio";

describe("tests FilterRadio", () => {
  const props = {
    value: "asc",
    label: "Ascending",
    handleChange: jest.fn(),
    checked: false,
  };
  it("should render component", async () => {
    // Arrange
    const tree = renderer.create(<FilterRadio {...props} />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });

  describe("display", () => {
    it("should display filter radio", async () => {
      // Act

      render(<FilterRadio {...props} />);

      // Assert
      expect(screen.getByText(props.label)).toBeInTheDocument();
      expect(screen.getByTestId(props.value).checked.toString()).toEqual(
        "false"
      );
    });
  });
});
