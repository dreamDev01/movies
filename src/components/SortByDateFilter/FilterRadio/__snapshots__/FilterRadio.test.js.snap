// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests FilterRadio should render component 1`] = `
<div
  className="radio"
>
  <input
    checked={false}
    data-testid="asc"
    onChange={[MockFunction]}
    type="radio"
    value="asc"
  />
  Ascending
</div>
`;
